visionmetrics/detection/mean_ap.py

```diff
@@ -34,14 +34,19 @@ def __init__(self, box_format='xyxy', **kwargs):
         if box_format != 'xyxy':
             raise ValueError(f'Expected box format to be "xyxy", got {box_format}')
         super().__init__(box_format=box_format, **kwargs)
+        self._label_ids = set()
 
     def update(self, predictions: List[List[List[float]]], targets: List[List[List[float]]]) -> None:
         predictions, targets = self._preprocess(predictions, targets)
+        self._update_label_ids(predictions, targets)
         super().update(predictions, targets)
 
     def compute(self):
         result = super().compute()
-        return {k: v for k, v in result.items() if k in ['map', 'map_50', 'map_75', 'map_per_class']}
+        if self.class_metrics:
+            result['map_per_class'] = {sorted(self._label_ids)[i]: v for i, v in enumerate(result['map_per_class'])}
```

Comment:
this sort got run by len(result['map_per_class']) times

Reply:
thanks. updated